### PR TITLE
Channel switching.

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

Channel switching.

## TODO

- [ ] Tests
- [ ] Make it more DRY.

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/actions.js [hunk 1]
```diff
@@ -69,27 +69,75 @@ function _collectionState(data) {
...
 
 
+/*
+ * Returns a promise that fetches the current channel id.
+ */
+function _getCurrentChannelId() {
+  let currentChannelId = null;
+  return new Promise((resolve, reject) => {
+    const channelListPromise = _getChannelList();
+    Promise.all([channelListPromise])
```
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/actions.js]: No need to use `Promise.all` here, as you are only waiting on one promise.

Can refactor to:

```
channelListPromise.then((channelList) => {
```

**Comment by christianmemije** [comment]
@rtibbles If I call showExploreChannel instead of using route.go, the channel id is no appended to the url.

**Comment by rtibbles** [comment]
@christianmemije Yeah, we should still do the required navigation event, I just wanted to make sure that we weren't duplicating any logic between the two. Also, I assume that router.go ends up proxying to `showExploreChannel`, right?

**Comment by christianmemije** [comment]
@rtibbles Yes that is correct. I removed the unnecessary rootTopicId promises from the root actions.

**Comment by rtibbles** [comment]
Definitely looks a lot DRYer now. Will test locally shortly.

File: kolibri/plugins/learn/assets/src/actions.js [hunk 2]
```diff
@@ -205,8 +290,10 @@ function showLearnChannel(store, channelId) {
   store.dispatch('SET_PAGE_NAME', PageNames.LEARN_CHANNEL);
   store.dispatch('SET_CURRENT_CHANNEL', channelId);
   cookiejs.set('currentChannel', channelId);
+  ContentNodeResource.setChannel(channelId);
 
-  const recommendedPromise = ContentNodeResource.getCollection({ recommendations: '' }).fetch();
+  const recommendedPromise =
+  ContentNodeResource.getCollection({ recommendations: '' }).fetch({}, true);
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/actions.js]: I think this line should be indented

File: kolibri/plugins/learn/assets/src/actions.js [hunk 3]
```diff
@@ -147,6 +197,42 @@ function redirectToLearnChannel(store) {
...
+          store.dispatch('SET_CHANNEL_LIST', channelList);
+        })
+        .catch((error) => {
+          store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
+          store.dispatch('CORE_SET_PAGE_LOADING', false);
+        });
+    })
+    .catch((error) => {
+
+    });
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/actions.js:223]: If we're not doing anything with the error, we _probably_ shouldn't have this block.

Perhaps the inner error catching should be moved out to here?

File: kolibri/plugins/learn/assets/src/actions.js [hunk 4]
```diff
@@ -85,30 +102,62 @@ function _getCurrentChannel() {
...
+            currentChannelRootTopicId = channelList[channel].root_pk;
+            break;
+          }
+        }
+      }
+      resolve(currentChannelRootTopicId);
+    })
+    .catch((error) => {
+
+    });
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/actions.js:129]: empty catch blocks are risky because they silently swallow errors

**Pull request merged**